File: backend/app/schemas/movie.py
# app/schemas/movie.py

from pydantic import BaseModel, Field
from datetime import date, datetime


class MovieBase(BaseModel):
    title: str = Field(..., max_length=255)
    release_date: date | None = None
    director: str | None = None
    genre: str | None = None
    poster_url: str | None = None


class MovieCreate(MovieBase):
    pass


class MovieResponse(MovieBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/movie.ts
//==============================================
// 영화 엔티티 단일 책임
// FastAPI schemas/movie.py와 1:1 대응
// 사용하는 곳, movieApi.ts, MovieCard.tsx, 
// Home.tsx
//==============================================

export interface Movie {
  id: number;
  title: string;
  release_date: string | null;
  director: string | null;
  genre: string | null;
  poster_url: string | null;
  created_at: string;
}

export interface MovieCreate {
  title: string;
  release_date?: string;
  director?: string;
  genre?: string;
  poster_url?: string;
}
